### PR TITLE
[BREAKING] Rename ui5HomeDir to ui5DataDir in APIs

### PR DESCRIPTION
**Pull request by d3xter666**
BREAKING CHANGE:
Installers and Resolvers' argument `ui5HomeDir` is now renamed to `ui5DataDir`

JIRA: CPOUI5FOUNDATION-802
Relates to: https://github.com/SAP/ui5-tooling/issues/701

### CODE REVIEW

**Comment by RandomByte** [comment]
Please add paragraph  starting with `BREAKING CHANGE:` to the commit body, explaining which APIs are affected and how

File: test/lib/ui5framework/Openui5Resolver.integration.js [hunk 1]
```diff
@@ -191,7 +191,7 @@ test.serial("resolveVersion", async (t) => {
 			cwd: options?.cwd ?? defaultCwd
 		})));
 		t.true(pacote.packument.alwaysCalledWithMatch("@openui5/sap.ui.core", {
-			cache: path.join(options?.ui5HomeDir ?? defaultUi5HomeDir, "framework", "cacache")
+			cache: path.join(options?.ui5DataDir ?? defaultUi5HomeDir, "framework", "cacache")
```
[the remark this answers is not in the document]
Comment (d3xter666) [reply test/lib/ui5framework/Openui5Resolver.integration.js]: Done

File: test/lib/ui5framework/Openui5Resolver.integration.js [hunk 2]
```diff
@@ -129,7 +129,7 @@ test.serial("resolveVersion", async (t) => {
 		undefined,
 		{
 			cwd: path.join(fakeBaseDir, "custom-cwd"),
-			ui5HomeDir: path.join(fakeBaseDir, "custom-homedir", ".ui5")
```
[the remark this answers is not in the document]
Comment (d3xter666) [reply test/lib/ui5framework/Openui5Resolver.integration.js:132]: Done

File: test/lib/ui5framework/AbstractResolver.js [hunk 3]
```diff
@@ -95,47 +95,47 @@ test("AbstractResolver: Defaults 'cwd' to process.cwd()", (t) => {
 	const {MyResolver} = t.context;
 	const resolver = new MyResolver({
 		version: "1.75.0",
-		ui5HomeDir: "/ui5home"
+		ui5DataDir: "/ui5home"
```
[the remark this answers is not in the document]
Comment (d3xter666) [reply test/lib/ui5framework/AbstractResolver.js:98]: Done

**Review by RandomByte — APPROVED** [review]

**Pull request merged**